File: back/src/main/java/fr/apside/apsipoker/championship/model/Championship.java
package fr.apside.apsipoker.championship.model;

import fr.apside.apsipoker.user.model.PokerUser;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;


@Entity
@Table(name = "tournament")
public class Championship {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column
    private String name;

    @Column
    private Date startDate;

    @Column
    private Date endDate;

    @ManyToMany
    private List<PokerUser> participants = new ArrayList<>();

    @OneToMany(mappedBy = "championship", orphanRemoval = true, cascade = CascadeType.ALL)
    private List<Tournament> tournaments = new ArrayList<>();

    public Championship() {
    }

    public Championship(String name, Date startDate, Date endDate) {
        this.name = name;
        this.startDate = startDate;
        this.endDate = endDate;
    }

    // region Getters & Setters

    public Long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Date getStartDate() {
        return startDate;
    }

    public void setStartDate(Date startDate) {
        this.startDate = startDate;
    }

    public Date getEndDate() {
        return endDate;
    }

    public void setEndDate(Date endDate) {
        this.endDate = endDate;
    }

    public List<PokerUser> getParticipants() {
        return participants;
    }

    public void setParticipants(List<PokerUser> participants) {
        this.participants.clear();
        this.participants.addAll(participants);
    }

    public List<Tournament> getTournaments() {
        return tournaments;
    }

    public void setTournaments(List<Tournament> tournaments) {
        this.tournaments = tournaments;
    }

    // endregion
}

File: front/src/app/championship/model/championship.ts
export class Championship {
  id?: number;
  name: string;
  startDate: string;
  endDate: string;
  participants: Player[];
  tournaments: Tournament[];

  constructor() {
    this.participants = [];
    this.tournaments = [];
  }
}

export class Tournament {
  id?: number;
  date: string;
  participants: TournamentPlayer[];

  constructor() {
    this.participants = [];
  }
}

export class TournamentPlayer {
  id?: number;
  player: Player;
  position: number;
  points: number;
}

export class Player {
  id: number;
  username: string;
  email: string;
  firstName: string;
  lastName: string;
  agency: string;
  role: string;
}
